### ISSUE TITLE
understanding the math !

### ISSUE THREAD

**Issue by RaphaelRaphael** [MurrellGroup/MultinomialLogisticGrowth#3]
Hi,

It was a pleasure to use your code and estimate rates of newer variants, we did it by processing GISAID metadata so it can be done very quickly. It lead to this preprint : https://www.biorxiv.org/content/10.1101/2024.01.20.576353v1.full.pdf

I tried to figure a bit what is the rational behind the model:
I can see that model.C is 2 dimensional matrix of intercepts (one value per lineage and country),
However, the formula provided for K appears to depend on each pair of lineages and country (so is LxLxC instead of LxC)

I am sure there are some details I am missing. Could you help me ?

Thank you

**Comment by murrellb** [comment]
Glad it was useful!

The easiest way to understand the model is:
- for any given country, the logits for each lineage are, over time, modeled with a straight line. Ie. with a slope (that is shared between countries) and an intercept (that is specific to each country)
- for any given country, for any point in time, the frequencies of the lineages are computed with the softmax over the lineage logits

It sounds, and is, extremely simple, but despite the underlying model being just straight lines, it provides a surprisingly good fit to the frequency data:
<img width="1616" alt="image" src="https://github.com/MurrellGroup/MultinomialLogisticGrowth/assets/1152087/15cfa25e-e71e-4c76-810a-61cae11c0829">

The formula in the readme:
$$\frac{l_i(t)}{l_j(t)}=K \cdot e^{t(g_i-g_j)}$$
is just intended to clarify that the ratio of the frequencies of any two lineages varies exponentially with time multiplied by the difference in growth rate between the two lineages, which is a consequence of the softmax in the model, and is relatively sensible for a process where one thing has a constant growth advantage over another.

Does that help?

**Comment by RaphaelRaphael** [comment]
Thank you ! 

Yes it help. Still the K in the formula seems to be related to the C in the code and C is per country AND lineage.

My question is whether or not I can extract (maybe from the intercept ) the average across lineages of difference of time between countries. Let s say that we could characterize how constant is the direction of spread of lineages in US running the model on this specific region.

**Comment by murrellb** [comment]
"Still the K in the formula seems to be related to the C in the code and C is per country AND lineage."

Yes. As K is "determined by their intercepts", it is specific to a country and a pair of lineages.

"My question is whether or not I can extract (maybe from the intercept ) the average across lineages of difference of time between countries. Let s say that we could characterize how constant is the direction of spread of lineages in US running the model on this specific region."

Possibly, but there are some pitfalls. This model is specifically intended to (quickly) estimate relative growth rates, where you can aggregate data over all countries and obtain relatively precise point estimates of the growth rates. The intercepts, however, were "nuisance" parameters for our purposes, and for some regions/lineages (especially when there was very little data for that lineage in that region) they will be extremely poorly estimated. If you want to do anything that involves the intercepts, you really should quantify the uncertainty in their estimates, which is a bit non-trivial.

A better approach, if what you want is an average "delay in arrival" across lineages (if I understood correctly), might be to re-write the model so this quantity is directly estimated. It isn't completely obvious how best to go about this, but that is where I'd start.

**Comment by RaphaelRaphael** [comment]
Thanks a lot ! 
Ok so K is 3 dimensional, it s clear for me now.

Thank you !
